### ISSUE TITLE
Running on Windows

### ISSUE THREAD

**Issue by robross0606** [congruity/congruity#5]
I'm attempting to get this running under Windows and having a couple issues.  So far I've:

- Installed the official Python 3.9.1 64-bit  packages
- Used pip to install `six`, `wxPython` and `suds-jurko` packages.
- Run `python setup.py install` script which outputs:
```
running install
running build
running build_py
running install_lib
running install_data
running install_egg_info
running egg_info
writing congruity.egg-info\PKG-INFO
writing dependency_links to congruity.egg-info\dependency_links.txt
writing entry points to congruity.egg-info\entry_points.txt
writing top-level names to congruity.egg-info\top_level.txt
reading manifest file 'congruity.egg-info\SOURCES.txt'
writing manifest file 'congruity.egg-info\SOURCES.txt'
removing 'C:\Program Files\Python39\Lib\site-packages\congruity-20-py3.9.egg-info' (and everything under it)
Copying congruity.egg-info to C:\Program Files\Python39\Lib\site-packages\congruity-20-py3.9.egg-info
running install_scripts
Installing congruity-script.pyw script to C:\Program Files\Python39\Scripts
Installing congruity.exe script to C:\Program Files\Python39\Scripts
Installing mhgui-script.pyw script to C:\Program Files\Python39\Scripts
Installing mhgui.exe script to C:\Program Files\Python39\Scripts
The system cannot find the path specified.
```
The last line of "The system cannot find the path specified" leads me to believe there's an issue which I'm guessing is about `libConcord` since that is a requirement.  I have concordance installed, but not sure how/if it knows how to find it.

**Comment by robross0606** [comment]
I tried running `mhgui` anyway and got this:
![image](https://user-images.githubusercontent.com/2965467/104852469-2f4a4780-58c9-11eb-9df0-962288bde96b.png)

**Comment by swt2c** [comment]
Yeah, unfortunately it looks like you'll need to install the libconcord Python bindings manually.  You should be able to do so by downloading the concordance tarball (matching the concordance version you installed) and then go into libconcord/bindings/python and run `python setup.py install` from in there.

**Comment by robross0606** [comment]
Did that.  It didn't produce any errors on the setup, but still getting issue running mhgui or congruity:
![image](https://user-images.githubusercontent.com/2965467/104852787-dda2bc80-58ca-11eb-940a-37e0b5279006.png)
How do the bindings know where to find libconcord?  Can it use the dll files installed in the Windows version of Concordance or am I going to have recompile the whole package?

**Comment by swt2c** [comment]
Hmm.  For some reason, the bindings don't try to load the DLL named with the ABI version.  Can you try editing libconcord.py and modify that line 40 to reference the `libconcord-5.dll` instead of `libconcord.dll` ?

**Comment by robross0606** [comment]
No luck.
![image](https://user-images.githubusercontent.com/2965467/104853821-57d63f80-58d1-11eb-9d37-0be2d2feda06.png)
Will the bindings have problems if I'm using 64-bit Python and these are 32-bit libraries?

**Comment by robross0606** [comment]
I started to give up and do this on an Ubuntu 20.02 virtual machine, but that isn't working either.  Very similar error even after installing `libconcord-dev` apt package:
![image](https://user-images.githubusercontent.com/2965467/104854354-2a3ec580-58d4-11eb-95c7-8d92b6d38577.png)

**Comment by robross0606** [comment]
I finally got this working in an Ubuntu Virtual Machine, but hit smack into another issue which I'm logging as possible separate bug.

**Comment by swt2c** [comment]
> Will the bindings have problems if I'm using 64-bit Python and these are 32-bit libraries? Where is it expecting to find the dll file?

To be honest with you, I'm not sure how dll file finding works on Windows.  You might try specifying the full path to the dll file.

**Comment by swt2c** [comment]
> I finally got this working in an Ubuntu Virtual Machine, but hit smack into another issue which I'm logging as possible separate bug.

Heh, saw your earlier comment about Ubuntu and came here to read it but it seems you must have deleted it.  :P

**Comment by robross0606** [comment]
Yeah, evolving rapidly here.  I figured out that I needed to install `python3-libconcord` as well as `libconcord-dev`.

**Comment by swt2c** [comment]
On Ubuntu, you could have also just installed 'congruity' from the Ubuntu repos and it would have installed all of that for you.  :-)

**Comment by robross0606** [comment]
![image](https://user-images.githubusercontent.com/2965467/104854637-df25b200-58d5-11eb-83ad-f1939c2b7139.png)

**Comment by robross0606** [comment]
Sadly, doesn't look like Congruity ends up being able to do either of the things I was hoping to do:

- Change the icon on an activity.
- Remove physical button assignments that have been assigned.

I cannot believe either of these things is not available in MyHarmony itself.  The first one used to be there but they removed it.  The second doesn't appear to have been thought thru.  Once you assign a button an action, there's literally no way to UNASSIGN the button to nothing.  You can only reassign it to another action.

**Comment by swt2c** [comment]
Sorry.  :(  Please file another issue(s) with those items and I'll try to look into whether they may be possible with the backend.  Leave this one open for better Windows support.

**Comment by LeVraiRoiDHyrule** [comment]
> Yeah, evolving rapidly here. I figured out that I needed to install `python3-libconcord` as well as `libconcord-dev`.

Hi,
How did you install those on Windows ? I'm having the same issue with mhgui script, path not found. But I can't find how to install those. Where did you get them ?

Thanks for any answer
